src/fs_io.h

```diff
@@ -17,17 +17,28 @@
 
 class MemBuf;
 
-// POD
 class dread_ctrl
 {
+    MEMPROXY_CLASS(dread_ctrl);
 public:
-    int fd;
-    off_t offset;
-    int req_len;
-    char *buf;
-    int end_of_file;
-    DRCB *handler;
-    void *client_data;
+    dread_ctrl(int aFd, char *aBuf, int aLen, off_t aOffset, DRCB *aHandler, void *aData) :
+        fd(aFd),
+        offset(aOffset),
+        req_len(aLen),
+        buf(aBuf),
+        handler(aHandler),
+        client_data(aData)
+    {}
+    dread_ctrl(dread_ctrl &&) = delete; // no copying or moving of any kind
+    ~dread_ctrl() = default;
+
+    int fd = -1;
+    off_t offset = 0;
+    int req_len = 0;
+    char *buf = nullptr;
+    int end_of_file = 0;
+    DRCB *handler = {};
+    void *client_data = {};
```

Comment:
By default, please use nullptr to initialize raw pointers.

```suggestion
    DRCB *handler = nullptr;
    void *client_data = nullptr;
```

Alternatively, do not initialize data members that are supposed to be set by the class constructor. This approach allows the compiler to warn us if we forget to copy constructor parameter to a class data member.